### PR TITLE
Android 13 show once notification permission dialog

### PR DESCRIPTION
**Pull request by sujitacharya2005**
Resolves https://github.com/brave/brave-browser/issues/26930

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/onboarding/OnboardingPrefManager.java [hunk 1]
```diff
@@ -341,4 +343,20 @@ private long setTimeInMillis(int timeInMinutes) {
...
+     */
+    public boolean isNotificationPermissionEnablingDialogShown() {
+        return mSharedPreferences.getBoolean(PREF_NOTIFICATION_PERMISSION_ENABLING_DIALOG, false);
+    }
+
+    /**
+     * Sets the user preference for whether the Notification Permission Enabling dialog is shown.
+     */
+    public void setNotificationPermissionEnablingDialogShown(boolean isShown) {
+        SharedPreferences.Editor sharedPreferencesEditor = mSharedPreferences.edit();
```
Comment (deeppandya) [inline android/java/org/chromium/chrome/browser/onboarding/OnboardingPrefManager.java:358]: We should start using SharedPreference Manager for all pref updates.

**Review by deeppandya — COMMENTED** [review]
We need to refactor prefs here and update it with SharedPreferenceManager. Can you create an issue for it ?

**Review by deeppandya — APPROVED** [review]
LGTM. We will address refactoring part in upcoming issue.

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Samsung Galaxy S10+` running `Android 13` using the following build(s):

```
Brave | 1.48.9 Chromium: 108.0.5359.48 (Official Build) canary (32-bit)
--- | ---
Revision | 18ceeca0d99318e70c00d2e04d88aa55488b5c63-refs/branch-heads/5359@{#854}
OS | Android 13; Build/T1B3.221003.008
```

* ensured that the warning modals implemented via https://github.com/brave/brave-core/pull/15407#issue-1404507730 are only being displayed once.
* ensured that `Enable notifications` for rewards is being shown when enabling rewards and OS level notification is disabled

Created https://github.com/brave/brave-browser/issues/27002 after discussing with @deeppandya & @sujitacharya2005.

**Pull request merged**